packages/trader/src/App/Components/Elements/ContractAudit/contract-audit.tsx

```diff
@@ -40,18 +44,25 @@ const ContractAudit = ({
     const getSortedUpdateHistory = (history: TContractUpdateHistory) =>
         history.sort((a, b) => Number(b?.order_date) - Number(a?.order_date));
 
+    const requestUpdatedHistory = React.useCallback((id?: number) => {
+        if (!id) return;
+        WS.contractUpdateHistory(id).then((response: TResponse) => {
+            setUpdateHistory(getSortedUpdateHistory(response.contract_update_history));
+        });
+    }, []);
+
     React.useEffect(() => {
         if (!!contract_update_history.length && contract_update_history.length > update_history.length)
             setUpdateHistory(getSortedUpdateHistory(contract_update_history));
     }, [contract_update_history, update_history]);
 
+    React.useEffect(() => {
+        if (is_history_tab_active && current_lang_to_request_history) requestUpdatedHistory(contract_id);
+    }, [contract_id, is_history_tab_active, current_lang_to_request_history, requestUpdatedHistory]);
```

Comment:
Do we need `requestUpdatedHistory` in array with dep-s?

Reply:
in this case, it doesn't create any issues. it's wrapped in `useCallback`, so that it's created only once. I added it to deps in order to avoid adding a comment about `exhaustive-deps`.